pydash/pydash_app/dashboard/dashboard.py

```diff
@@ -20,10 +22,12 @@ def __init__(self, url, user_id):
         self.id = uuid.uuid4()
         self.url = url
         self.user_id = uuid.UUID(user_id)
-        self.endpoints = list()
-        self.endpoint_calls = list()  # list of unfiltered endpoint calls, for use with an aggregator.
+        self.endpoints = []
         self.last_fetch_time = None
 
+        self._endpoint_calls = []  # list of unfiltered endpoint calls, for use with an aggregator.
+        self._aggregator = Aggregator(self._endpoint_calls)
```

Comment:
Having the aggregator as a private field seems like a good idea. Having a public function to extract the dashboard's aggregator data would be helpful.

Reply:
Will do.

Reply:
There is another problem with this line: The aggregator will be initialized with a reference to exactly the same list as `dashboard._endpoint_calls`. 

Observe:
```python
> a = [1,2,3]
> b = a
> a.append(4)
> b
[1,2,3,4]
```

In this case this means that a `call` is added twice when line 63+64 are executed.

(That is this part:
```python
        self._endpoint_calls.append(endpoint_call)
        self._aggregator.add_endpoint_call(endpoint_call)
```

Reply:
Good point. This means that the aggregator should not have to update its internal list of endpoint_calls, or that we should initialise it with a copy of the original list (while the objects inside still refer to the original endpoint_calls). I'm not entirely sure which one would be more clear/elegant/python-esque.

Reply:
@Qqwy I'm aware of the reference, which is why the endpoint call isn't added to the list in `_aggregator.add_endpoint_call`. Or are you referring to something else?

Reply:
(pun not intended 😂)

Reply:
It seems I have not looked close enough to the aggregator implementation. It seems like it should work.
One welcome change is to rename the `_aggregator.add_endpoint_call` function to something like `_aggregator.process_endpoint_addition` to avoid confusion, as it currently sounds like this method adds endpoints.

Reply:
`process_endpoint_addition` sounds to me even more like it is about adding an endpoint. maybe `append_endpoint_call`?

Reply:
Anyhow: That issue is non-blocking and can be fixed after this PR is merged